command/agent/consul/script_test.go

```diff
@@ -0,0 +1,173 @@
+package consul
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/nomad/helper/testtask"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+func TestMain(m *testing.M) {
+	if !testtask.Run() {
+		os.Exit(m.Run())
+	}
+}
+
```

Comment:
Can you test:

* exit code 0
* exit code 1
* exit code 2
* Capturing output
* Unknown command